### DISCUSSION TITLE
Docstrings

### DISCUSSION THREAD

**Discussion by ericmjl** [aesara-devs/aeppl#13] [General]
I'm happy to work on docstrings if you find this task to be tedium. :smile:

Let me know, @brandonwillard!

**Comment by brandonwillard** [comment]
Of course we're open to docstrings; that would be much appreciated!

**Comment by ericmjl** [comment]
Would like to explore and see what you would consider to be highest priority at the moment.

For now, I have seen the following places where I might be able to add docstrings:

1. Module-level docstrings for `utils.py` and `logprob.py`, which are two places where I sort of have an idea of what's going on.
2. Docstrings for the log prob functions in `logprob.py`. They are apparently self-explanatory, though, so I'm not sure if docstrings are worth it here.

For the module-level docstrings for `utils.py` and `logprob.py`, I was thinking of adding more than just a single liner, which would be quite meaningless. Rather, I was thinking of putting in the _intent_ of the module's existence in there plus architectural considerations to think about before making changes to the file. Does this sound about right to you?

**Comment by brandonwillard** [comment]
Yes, that sounds great; however, I don't know the extent to which we can detail all of the intent and architecture, since&mdash;for at least the latter&mdash;that's still being worked-out.

**Comment by ericmjl** [comment]
Got it. What is your take on docstrings for the log prob functions, @brandonwillard? Necessary or unnecessary?

**Comment by brandonwillard** [comment]
Probably not necessary, because the parameters are just the ones used by the `RandomVariable`s, and explaining those is better done in the docstrings for the `RandomVariable`s themselves (see https://github.com/aesara-devs/aesara/issues/389 and https://github.com/aesara-devs/aesara/issues/466).

There might be a few notes worth adding about the exact choice of log-probabilities, but none that I can think of right now.

**Comment by ericmjl** [comment]
Thanks @brandonwillard. Let me comb through the log-prob implementations to see if there's anything noteworthy there. I'll start off with that.
